include/mbedtls/fsio.h

```diff
@@ -0,0 +1,246 @@
+#endif
+
+#if defined(MBEDTLS_FS_IO)
+/* If MBEDTLS_FS_IO is enabled then file IO functions should be made available
+ * via standard library or platform specific implementation. */
+
+
+/**
+ * Structure returned by mbedtls_stat().
+ *
+ * \note Future versions of the library may add more fields at the end of this
+ * 	 structure.
+ */
+typedef struct mbedtls_stat_tag
+{
+    uint16_t    type;   /** File type */
+/** A file that can be accessed as a stream, i.e. opened with mbedtls_fopen().
+ */
+#define MBEDTLS_FSIO_DT_FILE    0
+/** A directory that can be opened with mbedtls_opendir(). */
+#define MBEDTLS_FSIO_DT_DIR     1
+/** A directory entry that is recognized neither as a stream file nor as a
+ * directory. */
+#define MBEDTLS_FSIO_DT_OTHER   2
+} mbedtls_stat_t;
+
+/* Default and alternative implementation specific interfaces. */
+#if !defined(MBEDTLS_FS_IO_ALT)
+#include <stdio.h>
+#include <dirent.h>
+
+typedef FILE *  mbedtls_file_t;
+#define mbedtls_fread( buf, size, stream )      fread( buf, 1, size, stream )
+#define mbedtls_fwrite( buf, size, stream )     fwrite( buf, 1, size, stream )
+#define mbedtls_fgets       fgets
+#define mbedtls_fclose      fclose
+#define mbedtls_ferror      ferror
+#define mbedtls_fseek       fseek
+#define mbedtls_ftell       ftell
+#define MBEDTLS_SEEK_SET    SEEK_SET
+#define MBEDTLS_SEEK_CUR    SEEK_CUR
+#define MBEDTLS_SEEK_END    SEEK_END
+#define MBEDTLS_FILE_INVALID    NULL
+
+typedef DIR *   mbedtls_dir_t;
+#define mbedtls_opendir         opendir
+#define mbedtls_closedir        closedir
+#define MBEDTLS_DIR_INVALID     NULL
+
+#else /* !MBEDTLS_FS_IO_ALT */
+
+/**
+ * file handle.
+ *
+ */
+typedef int32_t mbedtls_file_t;
+#define MBEDTLS_FILE_INVALID    -1
+
+/**
+ * dir handle.
+ *
+ */
+typedef int32_t mbedtls_dir_t;
+#define MBEDTLS_DIR_INVALID     -1
+
+/**
+ * Definition of reference for seek offset used by mbedtls_fseek().
+ */
+#define MBEDTLS_SEEK_SET    0
+#define MBEDTLS_SEEK_CUR    1
+#define MBEDTLS_SEEK_END    2
+
+/**
+ * \brief          Read file. Follows standard C fread interface.
+ *
+ * \param ptr      Pointer to output buffer
+ * \param size     Size of output buffer.
+ * \param stream   File handle (mbedtls_file_t).
+ *
+ * \return         Number of items read.
+ */
+size_t mbedtls_fread( void *ptr, size_t size, mbedtls_file_t stream );
+
+/**
+ * \brief          Write file. Follows standard C fwrite interface.
+ *
+ * \param ptr      Pointer to input buffer
+ * \param size     Bytes to write.
+ * \param stream   File handle of type mbedtls_file_t.
+ *
+ * \return         Number of items written.
+ */
+size_t mbedtls_fwrite( const void *ptr, size_t size, mbedtls_file_t stream );
+
+/**
+ * \brief          Reads a line from file. Follows standard C fgets interface.
+ *
+ * \param s        Pointer to output buffer.
+ * \param size     Size of buffer.
+ * \param stream   File handle of type mbedtls_file_t.
+ *
+ * \return         returns s on success, and NULL on error or
```

Comment:
`returns` -> `Returns`

Reply:
Or just nothing, since the paragraph header already reads “Returns”.

Reply:
@mazimkhan Please remove 'returns' and split the documentation in multiple `\return` sections, and use `\p s` and `\c NULL`.